pkg/apis/node/validation/validation.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	genericvalidation "k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/util/diff"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/apis/node"
+)
+
+// ValidateRuntimeClass validates the RuntimeClass
+func ValidateRuntimeClass(rc *node.RuntimeClass) field.ErrorList {
+	allErrs := genericvalidation.ValidateObjectMeta(&rc.ObjectMeta, false, genericvalidation.NameIsDNSSubdomain, field.NewPath("metadata"))
+	allErrs = append(allErrs, ValidateRuntimeClassSpec(&rc.Spec, field.NewPath("spec"))...)
+	return allErrs
+}
+
+// ValidateRuntimeClassSpec validates the sink spec for audit
+func ValidateRuntimeClassSpec(spec *node.RuntimeClassSpec, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if spec.RuntimeHandler != nil && *spec.RuntimeHandler != "" {
```

Comment:
if it is truly optional, then "" should be invalid - it means someone went out of their way to specify the field with an invalid value.

OTOH, as per above, I don't think this should be optional, so "" is invalid by that metric, but would be field.Required() rather than field.Invalid()

Reply:
I thought it was considered best-practice to treat empty & nil as equivalent, because json? I agree that if we decide runtimeHandler should be required, we should also forbid the empty value.

Reply:
nil and empty are distinct for strings, just not for lists (slices) and maps